test/e2e/util.go

```diff
@@ -20,39 +22,14 @@ import (
 
 	configv1 "github.com/openshift/api/config/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	operatorv1alpha1 "github.com/openshift/external-dns-operator/api/v1alpha1"
 )
 
 type providerTestHelper interface {
-	ensureHostedZone(string) (string, []string, error)
-	deleteHostedZone(string) error
+	ensureHostedZone() (string, []string, error)
+	deleteHostedZone() error
 	platform() string
-	makeCredentialsSecret(namespace string) *corev1.Secret
-}
-
-func rootAWSCredentials(kubeClient client.Client) (string, string, error) {
-	secret := &corev1.Secret{}
-	secretName := types.NamespacedName{
-		Name:      "aws-creds",
-		Namespace: "kube-system",
-	}
-	if err := kubeClient.Get(context.TODO(), secretName, secret); err != nil {
-		return "", "", fmt.Errorf("failed to get credentials secret %s: %w", secretName.Name, err)
-	}
-	return string(secret.Data["aws_access_key_id"]), string(secret.Data["aws_secret_access_key"]), nil
-}
-
-func rootGCPCredentials(kubeClient client.Client) (string, error) {
-	secret := &corev1.Secret{}
-	secretName := types.NamespacedName{
-		Name:      "gcp-credentials",
-		Namespace: "kube-system",
-	}
-	if err := kubeClient.Get(context.TODO(), secretName, secret); err != nil {
-		return "", fmt.Errorf("failed to get credentials secret %s: %w", secretName.Name, err)
-	}
-	return string(secret.Data["service_account.json"]), nil
+	makeCredentialsSecret() *corev1.Secret
+	defaultExternalDNS(credsSecret *corev1.Secret) operatorv1alpha1.ExternalDNS
```

Comment:
I think `default` in the name made sense when it was defined in the single place: there was only 1 externaldns instance, so default one. Now when each provider returns its instance I think that just `externalDNS` is more logical.

Reply:
Done